docs/source/Getting-Started/How-Keystone-Works/Keystone-Basics.rst

```diff
@@ -0,0 +1,113 @@
+
+**Security Monitor (SM)** is an M-mode software with small TCB.
+SM provides an interface for managing the lifecycle of enclave, as well as for utilizing platform-specific features.
+The SM is crucial for the security guarantees of the TEE since it enforces the isolation boundary between the enclaves and the untrusted OS.
+
+**Enclaves** are trusted execution environments isolated from the untrusted OS or the other enclaves. Each enclave is given a private physical memory region which is exclusively accessible by the enclave itself. Each enclave consists of an user-level enclave application (eapp) and a supervisor-level *runtime*.
+
+**Enclave Application (eapp)** is the user-level application that will run in the enclave. One can build a custom eapp from the scratch, or just run an existing RISC-V executable in Keystone.
+
+**Runtime** is a S-mode software, which selectively serves the untrusted kernel's functionality such as system call, trap handling, virtual memory management and so on.
+
+Keystone Workflow
+-------------------------------
+
+.. figure:: /_static/images/keystone_workflow.png
+
+Keystone is a framework wherein both *platform provider* and *enclave developer* can customize their enclaves depending on their needs.
+This involves two separate workflow from each of the platform provider and the enclave developer. 
+
+Provisioning the Security Monitor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A **platform provider** is a party who provides a machine which can run Keystone enclaves. In
+*provisioning* stage, the platform provider compiles the security monitor and deploys it to the machine. 
+
+The platform provider 
+configures and builds the security monitor with target hardware
+(❶ and ❷).
+The platform provider then deploys the security monitor to the machine (❸).
+
+See :doc:`/Getting-Started/Tutorials/Security-Monitor-Platform-Build` to learn how to build and
+deploy the security monitor.
+
+Developing the Enclave
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A **enclave developer** develops an enclave with Keystone SDK.
+The developer builds the eapp, the host, and the runtime binaries, and package them together into
+a
+single file (❹).
+
+The enclave is sent to the untrusted remote machine and deployed (❺, ❻, and ❼).
+
+See :doc:`/Getting-Started/Tutorials/Build-Enclave-App-Hello-World` to learn how to build an
+enclave.
+
+Remote Attestation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The enclave is challenged at the beginning of the execution by a trusted remote verifier (❽).
+The remote verifier could be the developer herself, a trusted party, or a person who wants to run an
+enclave built by others.
+The remote attestation uses cryptographically-secure signatures and measurements to check if a
+legitimate security monitor was deployed on a trusted hardware and if the enclave is initialized as expected.
+
+See :doc:`/Getting-Started/Tutorials/Remote-Attestation` to learn how to implement remote
+attestation.
+
+Enclave Lifecycle
```

Comment:
The sections on Lifecycle should be similarly shortened as in attestation and
   remove references to specific details of free memory, SBI call
   names, etc

Reply:
got it.